src/operator/src/expr_factory.rs

```diff
@@ -312,14 +312,14 @@ pub fn column_schemas_to_defs(
     column_schemas: Vec<ColumnSchema>,
     primary_keys: &[String],
 ) -> Result<Vec<api::v1::ColumnDef>> {
-    let column_datatypes = column_schemas
+    let column_datatypes: Vec<(ColumnDataType, Option<ColumnDataTypeExtension>)> = column_schemas
         .iter()
         .map(|c| {
             ColumnDataTypeWrapper::try_from(c.data_type.clone())
                 .map(|w| w.datatype())
```

Comment:
Maybe we can move the wrapper (use into)
```suggestion
                .map(|w| w.into_parts())
```